packages/api/lib/granule-remove-from-cmr.js

```diff
@@ -38,17 +39,34 @@ const _removeGranuleFromCmr = async (granule) => {
 const unpublishGranule = async (knexOrTransaction, granule) => {
   const granuleModelClient = new models.Granule();
   const granulePgModel = new GranulePgModel();
+  const collectionPgModel = new CollectionPgModel();
 
   await _removeGranuleFromCmr(granule);
 
-  await granulePgModel.update(
-    knexOrTransaction,
-    { granule_id: granule.granuleId },
-    {
-      published: false,
-      cmr_link: undefined,
+  // If we cannot find a PG Collection for this granule,
+  // don't update the PG Granule, continue to update the Dynamo granule
+  try {
+    const collectionCumulusId = await collectionPgModel.getRecordCumulusId(
+      knexOrTransaction,
+      deconstructCollectionId(granule.collectionId)
+    );
+
+    await granulePgModel.update(
+      knexOrTransaction,
+      {
+        granule_id: granule.granuleId,
+        collection_cumulus_id: collectionCumulusId,
+      },
+      {
+        published: false,
+        cmr_link: undefined,
+      }
+    );
+  } catch (error) {
+    if (!(error instanceof RecordDoesNotExist)) {
+      throw error;
     }
-  );
+  }
```

Comment:
This logic is a little weird right now because this `unpublishGranule` function is still driven by the Dynamo Granule. I think that's probably right for this phase but once we've fully switched to PG records I think we'll want to throw an error here if we can't get the PG Collection (or, by extension, update the PG Granule).

Reply:
> I think we'll want to throw an error here if we can't get the PG Collection (or, by extension, update the PG Granule).

If that is the intent, which I think it should be, then the code is not doing that. Inside the `catch`, we're only re-throwing if the error is **not** `RecordDoesNotExist`. But a `RecordDoesNotExist` is what would be thrown if we can't get the PG collection. So I think we want to remove this `if` condition at least, but really I think we can remove the entire try/catch?

Reply:
Sorry, the code is actually doing what I'd intended but my comment was unclear. 

What I'm saying in that comment (which I'm not saying is right but is just what I'm saying) is "What we should do in phase 2, once the PG reads are implemented and Dynamo is no longer the source of truth, is throw an error if we can't get the PG Collection or update the Granule." In other words unpublishGranule should fail if we can't get the PG Collection. BUT I don't think we want to do that now. I think the logic as-written might be what we want because I think this could potentially be used in the case where we don't have a PG record because it hasn't been migrated?

If some other error is thrown because the database connection is refused for some reason, that should be thrown.

Reply:
OK, yeah. I think that makes sense

I was vaguely concerned about the possibility of introducing drift here where the Dynamo record gets updated but not the PG record, but in theory if the PG collection record lookup fails then we most likely don't even have a PG granule (since it requires a collection).

So I think we're good 👍